src/lnd.ts

```diff
@@ -0,0 +1,151 @@
+  NewAddressResponse,
+  OpenChannelRequest,
+  Peer,
+  WalletBalanceResponse
+} from "lnd-async";
+import {
+  LndClient,
+  lndConnect,
+  LndConnectParams,
+  LndGetInfoResponse
+} from "./index";
+
+export class Lnd {
+  public static async fromClient(client: LndClient): Promise<Lnd> {
+    return new Lnd(client);
+  }
+
+  public static async fromParams(params: LndConnectParams): Promise<Lnd> {
+    const client = await lndConnect(params);
+    return new Lnd(client);
+  }
+
+  private constructor(public readonly client: LndClient) {}
+
+  public async genSeed(): Promise<string[]> {
+    const { cipher_seed_mnemonic } = await this.client.genSeed({});
+    if (!cipher_seed_mnemonic) {
+      throw new Error("lnd did not return the mnemonic.");
+    }
+    return cipher_seed_mnemonic;
+  }
+
+  public async initWallet(
+    walletPassword: Buffer,
+    seed: string[]
+  ): Promise<void> {
+    await this.client.initWallet({
+      wallet_password: walletPassword,
+      cipher_seed_mnemonic: seed
+    });
+  }
+
+  public async sendPayment(
+    publicKey: Buffer,
+    satAmount: number,
+    paymentHash: Buffer,
+    finalCltvDelta: number
+  ): Promise<void> {
+    await this.client.sendPayment({
+      dest: publicKey,
+      amt: satAmount,
+      payment_hash: paymentHash,
+      final_cltv_delta: finalCltvDelta
+    });
+  }
+
+  public async addHoldInvoice(
+    satAmount: number,
+    secretHash: Buffer,
+    expiry: number,
+    memo: string
+  ): Promise<string> {
+    const resp = await this.client.addHoldInvoice({
+      value: satAmount,
+      hash: secretHash,
+      expiry,
+      memo
+    });
+
+    return resp.payment_request;
+  }
+
+  public async settleInvoice(preimage: Buffer): Promise<void> {
+    return this.client.settleInvoice({ preimage });
+  }
+
+  public async getInfo(): Promise<LndGetInfoResponse> {
+    return this.client.getInfo({});
+  }
+
+  public async newAddress(type: AddressType): Promise<NewAddressResponse> {
+    return this.client.newAddress({ type });
+  }
+
+  public async channelBalance(): Promise<ChannelBalanceResponse> {
+    return this.client.channelBalance({});
+  }
+
+  public async walletBalance(): Promise<WalletBalanceResponse> {
+    return this.client.walletBalance({});
+  }
+
+  public async connectPeer(pubkey: string, host: string): Promise<void> {
+    const addr = { pubkey, host };
+    return this.client.connectPeer({ addr });
+  }
+
+  public async listPeers(): Promise<Peer[]> {
+    return (await this.client.listPeers({})).peers;
+  }
+
+  public async listChannels(request: ListChannelsRequest): Promise<Channel[]> {
+    return (await this.client.listChannels(request)).channels;
+  }
+
+  public async openChannel(
+    request: OpenChannelRequest
+  ): Promise<{
+    transactionId: string;
+    transactionVout: number;
+  }> {
+    const openChannel = this.client.openChannel(request);
+
+    return openChannel.on("data", (status: ChannelOpenStatusUpdate) => {
```

Comment:
I am trying to return `transactionId` and `transactionVout` out of the `Lnd.openChannel()` method but I am not sure how I do that as I get the info from withing the `.on()` block.